full/Dockerfile

```diff
@@ -237,6 +237,35 @@ RUN curl -fsSL https://pkgs.tailscale.com/stable/ubuntu/focal.gpg | sudo apt-key
     && apt-get update \
     && apt-get install -y tailscale
 
+### Install nix ###
+LABEL dazzle/layer=tool-nix
+LABEL dazzle/test=tests/tool-nix.yaml
+ENV NIX_VERSION=2.3.14
+# Dazzle does not rebuild a layer until one of its lines are changed. Increase this counter to rebuild this layer.
+ENV TRIGGER_REBUILD=1
+
+USER root
+RUN addgroup --system nixbld \
+  && adduser gitpod nixbld \
+  && for i in $(seq 1 30); do useradd -ms /bin/bash nixbld$i && adduser nixbld$i nixbld; done \
```

Comment:
Hey George, what exactly is happening in this line? Why are we iterating 30 times

Reply:
This creates the users that nix uses to compile software if the binary cache is a cache miss https://nixos.org/manual/nix/stable/installation/multi-user.html#setting-up-the-build-users